arrows/viscl/register_algorithms.cxx

```diff
@@ -33,32 +33,68 @@
  * \brief VisCL algorithm registration function implementation
  */
 
-#include "register_algorithms.h"
+#include <arrows/viscl/kwiver_algo_viscl_export.h>
+#include <vital/algo/algorithm_factory.h>
 
-#include <arrows/algorithm_plugin_interface_macros.h>
 #include <arrows/viscl/convert_image.h>
 #include <arrows/viscl/detect_features.h>
 #include <arrows/viscl/extract_descriptors.h>
 #include <arrows/viscl/match_features.h>
-#include <vital/registrar.h>
 
 
 namespace kwiver {
 namespace arrows {
 namespace vcl {
 
-/// Register VisCL algorithm implementations with the given or global registrar
-int register_algorithms( vital::registrar &reg )
+  extern "C"
+KWIVER_ALGO_VISCL_EXPORT
+void
+register_factories( kwiver::vital::plugin_loader& vpm )
 {
-  REGISTRATION_INIT( reg );
+  static auto const module_name = std::string( "algorithms.core" );
```

Comment:
"arrows.viscl"?  Note, for this one the "core" part definitely needs to be "viscl"